src/button-group/item-element.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { forwardRef } from 'react';
+import { ButtonGroupProps } from './interfaces.js';
+import { CancelableEventHandler, ClickDetail } from '../internal/events/index.js';
+import IconButtonItem from './icon-button-item.js';
+import MenuDropdownItem from './menu-dropdown-item.js';
+
+const ItemElement = forwardRef(
+  (
+    {
+      item,
+      dropdownExpandToViewport,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.Item;
+      dropdownExpandToViewport?: boolean;
+      onItemClick?: CancelableEventHandler<ButtonGroupProps.ItemClickDetails | ClickDetail>;
+    },
+    ref: React.Ref<HTMLDivElement>
+  ) => {
+    return (
+      <>
```

Comment:
nit: you can remove the fragment in favour of having a switch or if/else in the component's function body

Reply:
Using switch now